File: backend/src/main/java/ch/egli/streef/track/TrackDto.java
package ch.egli.streef.track;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import ch.egli.streef.trackpoint.TrackPointDto;
import lombok.Data;

@Data
public class TrackDto {
    private Long id;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private List<TrackPointDto> points = new ArrayList<>();
    private String notes;
    private double distanceKm;
    private double avgPaceMinPerKm;
    private LocalDateTime createdAt;
}

File: frontend/src/app/track/track.dto.ts
import { TrackPointDto } from './track-point.dto';

export interface TrackDto {
    id: number;
    startTime: string; // ISO string
    endTime: string;
    points: TrackPointDto[];
    notes: string;
    distanceKm: number;
    avgPaceMinPerKm: number;
    createdAt: string;
}
